request.go

```diff
@@ -981,3 +1022,80 @@ func (req *ExecuteRequest) Args(args interface{}) *ExecuteRequest {
 func (req *ExecuteRequest) Body(res SchemaResolver, enc *msgpack.Encoder) error {
 	return fillExecute(enc, req.expr, req.args)
 }
+
+// StickyRequest is an interface that provides the info about a Connection
+// the request belongs to.
+type StickyRequest interface {
+	Conn() *Connection
+}
+
+type PrepareRequest struct {
+	baseRequest
+	expr string
+}
+
+// NewPrepareRequest returns a new empty PrepareRequest.
+func NewPrepareRequest(expr string) *PrepareRequest {
+	req := new(PrepareRequest)
+	req.requestCode = PrepareRequestCode
+	req.expr = expr
+	return req
+}
+
+// Body fills an encoder with the execute request body.
+func (req *PrepareRequest) Body(res SchemaResolver, enc *msgpack.Encoder) error {
+	return fillPrepare(enc, req.expr)
+}
+
+type UnprepareRequest struct {
+	StickyRequest
+	baseRequest
+	stmt *PreparedStatement
+}
+
+// NewUnprepareRequest returns a new empty UnprepareRequest.
+func NewUnprepareRequest(stmt *PreparedStatement) *UnprepareRequest {
+	req := new(UnprepareRequest)
+	req.requestCode = PrepareRequestCode
+	req.stmt = stmt
+	return req
+}
+
+func (req *UnprepareRequest) Conn() *Connection {
+	return req.stmt.Conn
+}
+
+// Body fills an encoder with the execute request body.
+func (req *UnprepareRequest) Body(res SchemaResolver, enc *msgpack.Encoder) error {
+	return fillUnprepare(enc, *req.stmt)
+}
+
+type ExecutePreparedRequest struct {
+	StickyRequest
+	baseRequest
+	stmt *PreparedStatement
+	args interface{}
+}
+
+// NewPreparedExecuteRequest returns a new empty preparedExecuteRequest.
+func NewExecutePreparedRequest(stmt *PreparedStatement) *ExecutePreparedRequest {
+	req := new(ExecutePreparedRequest)
+	req.requestCode = ExecuteRequestCode
+	req.stmt = stmt
+	req.args = []interface{}{}
+	return req
+}
+
+func (req *ExecutePreparedRequest) Conn() *Connection {
+	return req.stmt.Conn
+}
```

Comment:
It seams to me that we can use an existing `ExecuteRequest` here. Assuming that there are requests with a dual nature in our API: it could have or no Connection.
```Go
if req.Conn() != nil {
    // have a connection
}
```

It would be easier on the user-side. But I'm ok from the code-side. Be free to do what you feel.